components/date-time/time.js

```diff
@@ -8,87 +8,158 @@ import moment from 'moment';
+	constructor() {
+		super( ...arguments );
+		this.state = {
+			hours: '',
+			minutes: '',
+			am: true,
+			date: null,
+		};
+		this.updateHours = this.updateHours.bind( this );
+		this.updateMinutes = this.updateMinutes.bind( this );
+		this.onChangeHours = this.onChangeHours.bind( this );
+		this.onChangeMinutes = this.onChangeMinutes.bind( this );
+	}
 
-	const minutes = selected.format( 'mm' );
-	const am = selected.format( 'A' );
-	const hours = selected.format( is12Hour ? 'hh' : 'HH' );
+	componentDidMount() {
+		this.syncState( this.props );
+	}
 
-	const updateHours = ( event ) => {
-		const value = parseInt( event.target.value, 10 );
+	componentWillReceiveProps( nextProps ) {
+		const { currentTime, is12Hour } = nextProps;
+		if (
+			currentTime !== this.props.currentTime ||
+			is12Hour !== this.props.is12Hour
+		) {
+			this.syncState( nextProps );
+		}
+	}
+
+	syncState( { currentTime, is12Hour } ) {
+		const selected = currentTime ? moment( currentTime ) : moment();
+		const minutes = selected.format( 'mm' );
+		const am = selected.format( 'A' );
+		const hours = selected.format( is12Hour ? 'hh' : 'HH' );
+		const date = currentTime ? moment( currentTime ) : moment();
+		this.setState( { minutes, hours, am, date } );
+	}
+
+	updateHours() {
+		const { is12Hour, onChange } = this.props;
+		const { am, hours, date } = this.state;
+		const value = parseInt( hours, 10 );
 		if (
 			! isInteger( value ) ||
 			( is12Hour && ( value < 1 || value > 12 ) ) ||
 			( ! is12Hour && ( value < 0 || value > 23 ) )
 		) {
+			this.syncState( this.props );
 			return;
 		}
 
 		const newDate = is12Hour ?
-			selected.clone().hours( am === 'AM' ? value % 12 : ( ( ( value % 12 ) + 12 ) % 24 ) ) :
-			selected.clone().hours( value );
-		onChange( newDate );
-	};
+			date.clone().hours( am === 'AM' ? value % 12 : ( ( ( value % 12 ) + 12 ) % 24 ) ) :
+			date.clone().hours( value );
+		this.setState( { date: newDate } );
+		const formattedDate = newDate.format( TIMEZONELESS_FORMAT );
+		onChange( formattedDate );
+	}
 
-	const updateMinutes = ( event ) => {
-		const value = parseInt( event.target.value, 10 );
+	updateMinutes() {
+		const { onChange } = this.props;
+		const { minutes, date } = this.state;
+		const value = parseInt( minutes, 10 );
 		if ( ! isInteger( value ) || value < 0 || value > 59 ) {
+			this.syncState( this.props );
 			return;
 		}
-		const newDate = selected.clone().minutes( value );
-		onChange( newDate );
-	};
+		const newDate = date.clone().minutes( value );
+		this.setState( { date: newDate } );
+		const formattedDate = newDate.format( TIMEZONELESS_FORMAT );
+		onChange( formattedDate );
+	}
 
-	const setAM = () => {
-		if ( am === 'AM' ) {
-			return;
-		}
-		const newDate = selected.clone().hours( parseInt( hours, 10 ) % 12 );
-		onChange( newDate );
-	};
+	updateAmPm( value ) {
+		return () => {
+			const { onChange } = this.props;
+			const { am, date, hours } = this.state;
+			if ( am === value ) {
+				return;
```

Comment:
Does this ever fire, or is it just for sanity here?

Reply:
It does fire when you hit the "toggled" button

Reply:
Ah cool, thanks for clarifying. It wasn't clear to me and it's not covered by the tests either, so I thought to ask :-)